### PR TITLE
avoid creating twice the temp table in dynamic insert overwrite

### LINKED ISSUE #810: [ADAP-674] [Bug] dbt_tmp is created twice with copy_partitions:true
### Is this a new bug in dbt-bigquery?

- [X] I believe this is a new bug in dbt-bigquery
- [X] I have searched the existing issues, and I could not find an existing issue for this bug

### Current Behavior

when using copy_partitions: true in partition_by parameter, it is creating dbt_tmp table twice before performing copy step

### Expected Behavior

it should only run dbt_tmp once

### Steps To Reproduce

model: test
```
  config(
    materialized='incremental',
    partition_by={
      "field": 'day_pt',
      "data_type": "datetime",
      "granularity": 'day',
      "copy_partitions": true
    }
  )
}}


SELECT
  DATETIME('2023-01-01') AS day_pt
```

first run: `dbt run --select test`

```
create or replace table `discord-pada-reporting`.`dbt_testing`.`test__chris_dong`
    partition by datetime_trunc(day_pt, day)


    OPTIONS(
      description=""""""
    )
    as (


SELECT
  DATETIME('2023-01-01') AS day_pt
    );
```

second run: `dbt run --select test`

```
  create or replace table `discord-pada-reporting`.`dbt_testing`.`test__chris_dong__dbt_tmp`
    partition by datetime_trunc(day_pt, day)


    OPTIONS(
      description="""""",

      expiration_timestamp=TIMESTAMP_ADD(CURRENT_TIMESTAMP(), INTERVAL 12 hour)
    )
    as (


SELECT
  DATETIME('2023-01-01') AS day_pt
    );
 ```
 is executed twice

### Relevant log output

_No response_

### Environment

```markdown
- OS:
- Python:
- dbt-core: 1.5.0
- dbt-bigquery: 1.5.0
```

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by Kayrnt**
resolves #810

### Problem

This PR aims to fix a case where a query is run twice as we create a temp table twice for following configuration:
`copy_partitions: true`, `incremental_strategy: "insert_overwrite"` and `on_schema_change != 'ignore' or language == 'python'`.

### Solution

As `copy_partitions` option requires a temp table.
We need to keep the logic in case `on_schema_change != 'ignore' or language == 'python'` is false, then we can use `tmp_relation_exists` flag to do so.

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] This PR has no interface changes (e.g. macros, cli, logs, json artifacts, config files, adapter interface, etc) or this PR has already received feedback and approval from Product or DX

### CODE REVIEW

**Comment by colin-rogers-dbt** [comment]
Is this change covered under the existing incremental tests?

**Comment by github-christophe-oudar** [comment]
> Is this change covered under the existing incremental tests?

No, not for all schema changes ([ignore is implicitely tested](https://github.com/dbt-labs/dbt-bigquery/blob/main/tests/functional/adapter/incremental/incremental_strategy_fixtures.py#L179-L191)), we could add some in https://github.com/dbt-labs/dbt-bigquery/blob/main/tests/functional/adapter/incremental/test_incremental_on_schema_change.py

**Comment by github-christophe-oudar** [comment]
I added a test but didn't have the time to check it during my lunch break 🙈

**Review by colin-rogers-dbt — APPROVED** [review]

**Comment by colin-rogers-dbt** [comment]
@Fleid what do we think about backporting this to a 1.6 patch?

File: dbt/include/bigquery/macros/materializations/incremental_strategy/insert_overwrite.sql [hunk 1]
```diff
@@ -19,7 +19,9 @@
 {% macro bq_copy_partitions(tmp_relation, target_relation, partitions, partition_by) %}
 
   {% for partition in partitions %}
-    {% if partition_by.granularity == 'hour' %}
+    {% if partition_by.data_type == 'int64' %}
+      {% set partition = partition | as_text %}
```
Comment (github-christophe-oudar) [inline dbt/include/bigquery/macros/materializations/incremental_strategy/insert_overwrite.sql:23]: For the record, this is fixing copy_partitions setting for `data_type: "int64"` (which now tested)

**Comment by github-christophe-oudar** [comment]
> @Fleid what do we think about backporting this to a 1.6 patch?
There is at least 2 persons interested if not much more, I'd fully support a 1.6.1 patch for that PR, solid under the hood gain for all dbt-bigquery users 👍

**Pull request merged**